src/adc.rs

```diff
@@ -719,25 +719,25 @@ where
         // Set the sample time for the channel
         match channel {
             #[cfg(feature = "gpio-f373")]
-            channel::Id::Zero => self.adc.smpr1.read().smp0().variant().into(),
-            channel::Id::One => self.reg.smpr1.read().smp1().variant().into(),
-            channel::Id::Two => self.reg.smpr1.read().smp2().variant().into(),
-            channel::Id::Three => self.reg.smpr1.read().smp3().variant().into(),
-            channel::Id::Four => self.reg.smpr1.read().smp4().variant().into(),
-            channel::Id::Five => self.reg.smpr1.read().smp5().variant().into(),
-            channel::Id::Six => self.reg.smpr1.read().smp6().variant().into(),
-            channel::Id::Seven => self.reg.smpr1.read().smp7().variant().into(),
-            channel::Id::Eight => self.reg.smpr1.read().smp8().variant().into(),
-            channel::Id::Nine => self.reg.smpr1.read().smp9().variant().into(),
-            channel::Id::Ten => self.reg.smpr2.read().smp10().variant().into(),
-            channel::Id::Eleven => self.reg.smpr2.read().smp11().variant().into(),
-            channel::Id::Twelve => self.reg.smpr2.read().smp12().variant().into(),
-            channel::Id::Thirteen => self.reg.smpr2.read().smp13().variant().into(),
-            channel::Id::Fourteen => self.reg.smpr2.read().smp14().variant().into(),
-            channel::Id::Fifteen => self.reg.smpr2.read().smp15().variant().into(),
-            channel::Id::Sixteen => self.reg.smpr2.read().smp16().variant().into(),
-            channel::Id::Seventeen => self.reg.smpr2.read().smp17().variant().into(),
-            channel::Id::Eighteen => self.reg.smpr2.read().smp18().variant().into(),
+            channel::Id::Zero => self.adc.smpr1().read().smp0().variant().into(),
```

Comment:
If you specify `channel::Id::Zero` as `0` you could write it shorter:
```rust
let ch = channel as u8 
if ch < 9 {
    self.adc.smpr1.read().smp(ch).variant().into()
} else {
    self.adc.smpr1.read().smp(ch - 9).variant().into()
}
```

Reply:
Like this a8f8835?

Reply:
They dont seem to be array-ified

Reply:
Sorry. You are right. I've not finished this yet.

Reply:
No problem :)